components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -3512,6 +3517,111 @@ private void deleteRequestPathAuthenticators(int applicationID, Connection conne
         }
     }
 
+    /**
+     * Get trusted app metadata of the application.
+     *
+     * @param applicationId Application ID.
+     * @param connection    Database connection.
+     * @param tenantID      Tenant ID.
+     * @return trustedAppMetadata Trusted app configurations.
+     * @throws IdentityApplicationManagementException If an error occurs while retrieving trusted app configurations.
+     */
+    private SpTrustedAppMetadata getSpTrustedAppMetadata(int applicationId, Connection connection, int tenantID)
+            throws IdentityApplicationManagementException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Retrieving trusted app configurations for application: " + applicationId);
+        }
+        SpTrustedAppMetadata spTrustedAppMetadata = new SpTrustedAppMetadata();
+
+        try (PreparedStatement loadAppConfigs = connection
+                    .prepareStatement(ApplicationMgtDBQueries.LOAD_TRUSTED_APPS_BY_APP_ID)) {
+            loadAppConfigs.setInt(1, applicationId);
+            loadAppConfigs.setInt(2, tenantID);
+
+            try (ResultSet appConfigResultSet = loadAppConfigs.executeQuery()) {
+                // There should be maximum two entries for each service provider. One for Android and one for iOS.
+                while (appConfigResultSet.next()) {
+                    String platformType = appConfigResultSet.getString(1);
+                    if (ANDROID.equals(platformType)) {
+                        spTrustedAppMetadata.setAndroidPackageName(appConfigResultSet.getString(2));
+                        spTrustedAppMetadata.setAndroidThumbprints(appConfigResultSet.getString(3));
+                    } else if (IOS.equals(platformType)) {
+                        spTrustedAppMetadata.setAppleAppId(appConfigResultSet.getString(2));
+                    }
+                    spTrustedAppMetadata.setIsFidoTrusted(appConfigResultSet.getBoolean(4));
+                }
+            }
+        } catch (SQLException e) {
+            throw new IdentityApplicationManagementException("Error while retrieving trusted app configurations.", e);
+        }
+        return spTrustedAppMetadata;
+    }
+
+    /**
+     * Update trusted app metadata of the application.
+     *
+     * @param applicationId  Application ID.
+     * @param trustedAppMetadata Trusted app configurations.
+     * @param connection     Database connection.
+     * @throws IdentityApplicationManagementException If an error occurs while updating trusted app configurations.
+     */
+    private void updateSpTrustedAppMetadata(int applicationId, SpTrustedAppMetadata trustedAppMetadata,
+                                            Connection connection) throws IdentityApplicationManagementException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Adding trusted app configurations for application: " + applicationId);
+        }
+        int tenantID = CarbonContext.getThreadLocalCarbonContext().getTenantId();
```

Comment:
Instead of resolving the tenantID here lets pass it in as a param. The place where this is used already has the tenantId.